tests/apps/test_similarity.py

```diff
@@ -222,142 +223,224 @@ def test_event_cardinality(photons, max_count, modes, expected):
+
+    def test_invalid_photon_number(self):
+        """Test if function raises a ``ValueError`` when a photon number below zero is specified"""
+        g = nx.complete_graph(5)
+        with pytest.raises(ValueError, match="Photon number must not be below zero"):
+            similarity._prob_event_exact(g, -1, 2)
+
+    def test_low_count(self):
+        """Test if function raises a ``ValueError`` if ``max_count_per_mode`` is negative."""
+        g = nx.complete_graph(5)
+        with pytest.raises(
+            ValueError, match="Maximum number of photons per mode must be non-negative"
+        ):
+            similarity._prob_event_exact(g, 2, max_count_per_mode=-1)
+
+    def test_prob_vacuum_event(self):
+        """Tests if the function gives the right probability for an event with zero photons when
+        the GBS device has been configured to have zero mean photon number."""
+        graph = nx.complete_graph(5)
+        assert similarity._prob_event_exact(graph, 0, 0, 0) == 1.0
+
+    @pytest.mark.parametrize("k", [3, 5, 7, 9])
+    @pytest.mark.parametrize("nmax", [1, 2])
+    def test_odd_photon_numbers(self, k, nmax):
+        """Test if function returns zero probability for odd number of total photons."""
+        graph = nx.complete_graph(10)
+        assert similarity._prob_event_exact(graph, k, nmax) == 0.0
+
+
 class TestProbOrbitMC:
-    """Tests for the function ``strawberryfields.apps.similarity.prob_orbit_mc.``"""
+    """Tests for the function ``strawberryfields.apps.similarity._prob_orbit_mc.``"""
 
     def test_invalid_samples(self):
         """Test if function raises a ``ValueError`` when a number of samples less than one is
         requested."""
-        g = nx.complete_graph(10)
+        g = nx.complete_graph(5)
         with pytest.raises(ValueError, match="Number of samples must be at least one"):
-            similarity.prob_orbit_mc(g, [1, 1, 1, 1], samples=0)
+            similarity._prob_orbit_mc(g, [1, 1, 1, 1], mc_samples=0)
 
     def test_invalid_n_mean(self):
         """Test if function raises a ``ValueError`` when the mean photon number is specified to
         be negative."""
-        g = nx.complete_graph(10)
+        g = nx.complete_graph(5)
         with pytest.raises(ValueError, match="Mean photon number must be non-negative"):
-            similarity.prob_orbit_mc(g, [1, 1, 1, 1], n_mean=-1)
+            similarity._prob_orbit_mc(g, [1, 1, 1, 1], n_mean=-1)
 
     def test_invalid_loss(self):
         """Test if function raises a ``ValueError`` when the loss parameter is specified outside
         of range."""
-        g = nx.complete_graph(10)
-        with pytest.raises(ValueError, match="Loss parameter must take a value between zero and"):
-            similarity.prob_orbit_mc(g, [1, 1, 1, 1], loss=2)
+        g = nx.complete_graph(5)
+        with pytest.raises(
+            ValueError, match="Loss parameter must take a value between zero and one"
```

Comment:
Note that the match doesn't have to be exact, it can be just the first few words - it should be enough to uniquely identify the error.

Reply:
oh nice! I did not know this :)